applications/plugins/OpenCTMPlugin/CMakeLists.txt

```diff
@@ -1,23 +1,44 @@
 cmake_minimum_required(VERSION 3.1)
-projetc(OpenCTMPlugin)
+project(OpenCTMPlugin VERSION 0.1)
+
+find_package(SofaBase REQUIRED)
+find_package(OpenCTM REQUIRED)
+
+set(OPENCTM_SRC_DIR src/SofaOpenCTM)
 
 set(HEADER_FILES
-    OpenCTMExporter.h
-    OpenCTMLoader.h
-    config.h
+    ${OPENCTM_SRC_DIR}/config.h.in
+    ${OPENCTM_SRC_DIR}/OpenCTMExporter.h
+    ${OPENCTM_SRC_DIR}/OpenCTMLoader.h
 )
 
 set(SOURCE_FILES
-    OpenCTMExporter.cpp
-    OpenCTMLoader.cpp
-    initOpenCTMPlugin.cpp
+    ${OPENCTM_SRC_DIR}/OpenCTMExporter.cpp
+    ${OPENCTM_SRC_DIR}/OpenCTMLoader.cpp
+    ${OPENCTM_SRC_DIR}/initOpenCTMPlugin.cpp
 )
 
 set(README_FILES OpenCTMPlugin.txt)
 
-find_package(OpenCTM REQUIRED)
-
 add_library(${PROJECT_NAME} SHARED ${HEADER_FILES} ${SOURCE_FILES} ${README_FILES})
-include_directories("${OPENCTM_INCLUDE_PATH}")
-add_definitions("-DSOFA_BUILD_OPENCTMPLUGIN")
-target_linker_libraries(${PROJECT_NAME} SofaBaseTopology ${OPENCTM_LIBRARY})
+
+# Set define dllimport/dllexport mechanism on Windows.
+target_compile_definitions(${PROJECT_NAME} PRIVATE "-DSOFA_BUILD_OPENCTMPLUGIN")
+
+target_link_libraries(${PROJECT_NAME} SofaBaseTopology SofaOpenglVisual ${OPENCTM_LIBRARY})
+
+target_include_directories(${PROJECT_NAME} PUBLIC "$<BUILD_INTERFACE:${OPENCTM_INCLUDE_PATH}>")
```

Comment:
CMake good practice: 
This line is not be needed if `find_package(OpenCTM REQUIRED)` defines a clean target with something in INTERFACE_INCLUDE_DIRECTORIES. 
You can use `debug_print_target_properties(OpenCTM::OpenCTM)` to dump the target OpenCTM::OpenCTM (if such a target exists).

Reply:
the target doesn't exist. That lib is quiet old and there is no nice findOpenCTM, I was even considering moving the code inside the plugin...